CIP-0057/README.md

```diff
@@ -0,0 +1,414 @@
+---
+CIP: 57
+Title: Plutus Contract Blueprint
+Authors: KtorZ <matthias.benkort@cardanofoundation.org>, scarmuega <santiago@carmuega.me>
+Discussions-To: https://github.com/cardano-foundation/CIPs/pull/258
+Status: Draft 
+Type: Process
+Created: 2022-05-15
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This document specifies a language for documenting Plutus contracts in a machine-readable manner. This is akin to what [OpenAPI](https://swagger.io/specification) or [AsyncAPI](https://www.asyncapi.com/docs/specifications/v2.4.0) are for, documenting HTTP services and asynchronous services respectively. In a similar fashion, A Plutus contract has a binary interface which is mostly defined by its datum and redeemer. Since all interactions with contracts are done through transactions, we can also model _endpoints_ of a contract as transitions from a known state to another purely declaratively. 
+
+This document is therefore a meta-specification defining the vocabulary and validation rules with which one can specify a Plutus contract interface, a.k.a **Plutus contract blueprint**.
+
+## Motivation
+
+While publicly accessible, on-chain contracts are currently inscrutable. Ideally, one would want to get an understanding of transactions revolving around script executions. This is both useful to visualize and to control the evolution of a contract life-cycle; but also, as a user interacting with a contract, to ensure that one is authorizing a transaction to do what it's intended to. Having a machine-readable specification in the form of a JSON-schema makes it easier (or even, possible) to enable a wide variety of use cases from a single concise document, such as:
+
+- Code generators for serialization/deserialization of Contract's elements
+- Contract API Reference / Documentation, also automatically generated
+- Extra automated transaction validation layers 
+- Better wallet UI / integration with DApps
+- Automated Plutus-code scaffolding 
```

Comment:
- I'm not sure what this is supposed to mean, do you mean you want to describe the datums, redeemers, token names in question?
- Human readable? Why would this be better than writing something manually?
- Not sure what this means.
- Improving wallet UX makes sense, noto sure what you mean about integration with dapps.
- I'm not sure this is useful for that. You usually go the other way around.

Reply:
It seems like you aren't familiar with tools such as OpenAPI or AsyncAPI, which I suggest to familiarize yourself with. Although, this would likely only speak to you if you've ever interacted with an HTTP or async service -- or designed one. These kind of use cases are very common for client applications integrating with an API service. In this scenario, client applications are wallets, explorers or any other tool really that is interacting with the chain to a certain extend, and the service is the on-chain DApp itself. At present, most DApps are also building their own frontend and somewhat obfuscating their contract interface. This proposal aims to make it possible for anyone to implement a somewhat robust frontend for a specified DApp (though this isn't the only use case).